hook.php

```diff
@@ -469,30 +454,19 @@ function plugin_formcreator_hook_delete_ticket(CommonDBTM $item) {
 
    $id = $item->getID();
 
-   // find a formanswer linked to the ticket
+   // Update the formanswer's status (for cases where a form answer has several tickets)
    $formAnswer = new PluginFormcreatorFormAnswer();
-   $formAnswer->getFromDBByCrit([
-      'id' => new QuerySubQuery([
-         'SELECT' => 'items_id',
-         'FROM'   => Item_Ticket::getTable(),
-         'WHERE'  => [
-            'itemtype' => PluginFormcreatorFormAnswer::getType(),
-            'tickets_id' => $id,
-         ]
-      ])
-   ]);
-   if (!$formAnswer->isNewItem()) {
+   if ($formAnswer->getFromDbByTicket($id)) {
       $minimalStatus = $formAnswer->getAggregatedStatus();
       if ($minimalStatus === null) {
          // There is no more ticket in the form anwer
          $formAnswer->updateStatus(CommonITILObject::CLOSED);
       } else {
          $formAnswer->updateStatus($minimalStatus);
       }
-      return;
```

Comment:
Maybe thre is only 1 ticket associated to a the form answer. In such case, we must also delete the issue which represents the ticket. See PluginFormcreatorFormAnswer::createIssue(). I nthis method if there is only 1 generated ticket  then we create an issue ssociated to this ticket.